### ISSUE TITLE
Digit query parameter keys in long URLs get reset to 0

### ISSUE THREAD

**Issue by pxeger** [shlinkio/shlink#1155]
#### How Shlink is set-up

* Shlink Version: 2.8.0 (from Docker)
* PHP Version: 8.0.9 (Zend Engine v4.0.9)
* How do you serve Shlink: Docker image (`shlinkio/shlink:stable`)
* Database engine used: SQLite

#### Summary

URLs with query parameter keys that are made only of digits, such as`123` in `https://example.com/?123=foo`, end up being replaced with `0` (in that case redirected to `https://example.com/?0=foo`).

#### Current behavior

Redirect to `https://example.com/?0=foo`

#### Expected behavior

Redirect to `https://example.com/?123=foo`

#### How to reproduce

1. Shorten a link such as `https://example.com/?123=foo`.
2. Visit the shortened link
3. Observe that you end up being redirected to `https://example.com/?0=foo`

---

The link appears to be stored correctly in the database:

```
sqlite> select * from short_urls;
1|||https://example.com/?123=x|BYWgS|2021-08-15 08:42:14|||||||0|0
```

But when checking the redirect header:

```bash
$ curl -Iv 'curl -Iv http://localhost:8080/BYWgS'
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> HEAD /BYWgS HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.78.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 Found
HTTP/1.1 302 Found
< Location: https://example.com/?0=x
Location: https://example.com/?0=x
< Server: swoole-http-server
Server: swoole-http-server
< Connection: keep-alive
Connection: keep-alive
< Content-Type: text/html
Content-Type: text/html
< Date: Sun, 15 Aug 2021 08:51:29 GMT
Date: Sun, 15 Aug 2021 08:51:29 GMT

< 
* Connection #0 to host localhost left intact
```

the parameter has been reset to 0.

This issue is not present when the parameter contains any letters like `a123`. I thought this might be a PHP type confusion thing, but `0x123` does not do this. This does not seem to happen in any other component of the URL

**Comment by acelaya** [comment]
Yep, it's quite easy to reproduce. I'll take a look.

**Comment by acelaya** [comment]
I have just released v2.8.1, which fixes this issue.

You'll have to wait for this build to finish to get the latest docker image published https://github.com/shlinkio/shlink/actions/runs/1133120575. It'll take around 1h
